src/common/libflux/message.c

```diff
@@ -1597,6 +1597,9 @@ void flux_msg_fprint (FILE *f, const flux_msg_t *msg)
         return;
     }
     prefix = msgtype_shortstr (msg->type);
+    /* Timestamp
+     */
+    fprintf (f, "%s %.5f\n", prefix, timestamp);
```

Comment:
What about keeping the existing `flux_msg_fprint()` (since it is in the public API, and there isn't much use case for the timestamp when using the function publicly, at least that I can think of), and adding a private `flux_msg_fprint_ts()` which takes a timestamp, but doesn't print it if `timestamp < 0.`. Then `flux_msg_fprint()` could just be:
```c
void flux_msg_fprint (FILE *f, const flux_msg_t *msg)
{
    flux_msg_fprint_ts (f, msg, -1.);
}
```

`handle_trace()` could be the sole user of `flux_msg_fprint_ts()`?

Reply:
There don't seem to be any other users, but I'd be fine with this proposed change.

Reply:
How can you know there are no other users?

Reply:
I withdraw my prior statement. :-)